Tests/RevenueCatUITests/PaywallsV2/TakeScreenshot.swift

```diff
@@ -0,0 +1,172 @@
+        ) else {
+            fatalError("Failed to decode offerings data")
+        }
+
+        let offeringsWithPackages = offeringsResponse.offerings.map { offering in
+            return OfferingsResponse.Offering(
+                identifier: offering.identifier,
+                description: offering.description,
+                packages: packages.packages,
+                paywallComponents: offering.paywallComponents,
+                draftPaywallComponents: offering.draftPaywallComponents,
+                webCheckoutUrl: offering.webCheckoutUrl
+            )
+        }
+
+        let offeringsResponseWithPackages = OfferingsResponse(
+            currentOfferingId: offeringsResponse.currentOfferingId,
+            offerings: offeringsWithPackages,
+            placements: offeringsResponse.placements,
+            targeting: offeringsResponse.targeting,
+            uiConfig: offeringsResponse.uiConfig
+        )
+
+        let offerings = OfferingsFactory().createOfferings(from: [
+            "com.revenuecat.lifetime_product": .init(sk1Product: PreviewMock.Product(
+                price: 1.99,
+                unit: .week,
+                localizedTitle: "Liftime"
+            )),
+            "com.revenuecat.annual_product": .init(sk1Product: PreviewMock.Product(
+                price: 1.99,
+                unit: .year,
+                localizedTitle: "Annual"
+            )),
+            "com.revenuecat.semester_product": .init(sk1Product: PreviewMock.Product(
+                price: 1.99,
+                unit: .month,
+                localizedTitle: "6 Month"
+            )),
+            "com.revenuecat.quarterly_product": .init(sk1Product: PreviewMock.Product(
+                price: 1.99,
+                unit: .week,
+                localizedTitle: "3 Month"
+            )),
+            "com.revenuecat.bimonthly_product": .init(sk1Product: PreviewMock.Product(
+                price: 1.99,
+                unit: .week,
+                localizedTitle: "2 Month"
+            )),
+            "com.revenuecat.monthly_product": .init(sk1Product: PreviewMock.Product(
+                price: 1.99,
+                unit: .week,
+                localizedTitle: "Monthly"
+            ))
+        ], data: offeringsResponseWithPackages)
+
+        for offeringId in offerings!.all.keys {
+            let offering = offerings!.all[offeringId]!
+
+            let view = Self.createPaywall(offering: offering)
+                .frame(width: 450, height: 1000)
+            self.snapshotAndSave(view: view, size: CGSize(width: 450, height: 1000), filename: "\(offeringId).png")
+        }
+    }
+
+    func snapshotAndSave<V: View>(view: V, size: CGSize, filename: String) {
+        let image = view.asImage(wait: 0.5)
+
+        let url = FileManager.default.temporaryDirectory.appendingPathComponent(filename)
+        try? image.pngData()?.write(to: url)
+        print("Saved screenshot to: \(url)")
+    }
```

Comment:
Since we're in tests, I think I would make sure the tests are failing when something is not working as expected. So I would probable do this
```suggestion
    func snapshotAndSave<V: View>(view: V, size: CGSize, filename: String) throws {
        let image = view.asImage(wait: 0.5)

        let url = FileManager.default.temporaryDirectory.appendingPathComponent(filename)
        let data = try XCTUnwrap(image.pngData(), "Failed to convert snapshot image for to PNG data (filename: \(filename))")
        XCTAssertNoThrow(try data.write(to: url), "Failed to save screenshot to \(url)")
    }
```

Reply:
Oh neat! Can add that 😇